### PR TITLE
[Ray Clusters][docs] Create new Running Apps on Ray Clusters section

### PR DESCRIPTION
**Pull request by cadedaniel**
## Why are these changes needed?
This adds the structure [described here](https://docs.google.com/document/d/1jCToq9s2R8CYH1Ro4aftjZSYLLn1i2NF0AMfriszCFw/edit#heading=h.3s2ivpfx5sd3), namely adding a new section under `Ray Clusters` which is focused on running applications on Ray clusters.

I moved the content around best I could to unblock @stephanie-wang writing content around jobs.

## Related issue number

N/A

### CODE REVIEW

File: doc/source/_toc.yml [hunk 1]
```diff
@@ -283,7 +283,7 @@ parts:
                 - file: cluster/cluster_under_construction/ray-clusters-on-kubernetes/examples/ml-example.md
             - file: cluster/cluster_under_construction/ray-clusters-on-kubernetes/references
           - file: cluster/cluster_under_construction/ray-clusters-on-vms/index
-            title: Ray Clusters on VMs
+            title: Deploying a Ray Cluster on VMs
```
Comment (stephanie-wang) [inline doc/source/_toc.yml:286]:
```suggestion
            title: Deploy a Ray Cluster on VMs
```
Keeping the naming consistent with the other page.

File: doc/source/_toc.yml [hunk 2]
```diff
@@ -322,6 +320,8 @@ parts:
                 sections:
                   - file: cluster/cluster_under_construction/ray-clusters-on-vms/examples/setting-up-ray-cluster-for-production/todo
             - file: cluster/cluster_under_construction/ray-clusters-on-vms/references/index
+          - file: cluster/cluster_under_construction/running-applications-on-ray-clusters/index
+            title: Running Applications on Ray Clusters
```
Comment (stephanie-wang) [inline doc/source/_toc.yml:324]: Related to above, maybe "Run an Application on a Ray Cluster" instead (or change the other two to match this).

**Review by stephanie-wang — APPROVED** [review]

File: doc/source/_toc.yml (hunk 1 above)
Comment (cadedaniel) [reply doc/source/_toc.yml:286]: Nice catch.. I'm probably going too fast if I missed this

File: doc/source/_toc.yml (hunk 2 above)
Comment (cadedaniel) [reply doc/source/_toc.yml:324]: Hm, I think "Deploying" and "Running" are better. Will change them to be consistent

**Pull request merged**